include/lang/py/py.js

```diff
@@ -813,3 +813,102 @@ function runtime_get_syntaxError_thrower(compilationError) {
 
     return syntaxError
 }
+
+// FFI
+var host_eval = eval;
+
+var _foreign_counter = 0;
+function gen_foreign() { return "___f" + _foreign_counter++; }
+
+// Host type predicates available to pyinterp
+// is_host_foo => foo is the Python type name
+function is_host_bool(obj) {
+    return typeof obj === "boolean";
+}
+function is_host_str(obj) {
+    return typeof obj === "string";
+}
+function is_host_function(obj) {
+    return typeof obj === "function";
+}
+function is_host_float(obj) {
+    return (typeof obj === "number") && !Number.isInteger(obj);
+}
+function is_host_list(obj) {
+    return obj instanceof Array;
+}
+function is_host_dict(obj) {
+    return obj instanceof Object;
+}
+function is_host_int(obj) {
+    return Number.isInteger(obj);
+}
+
+// Convert a pyinterp object to a JS object
+function py2host(obj) {
+    var type = (t) => pyinterp.om_isinstance(obj, t);
+
+    if (type(pyinterp.class_int)) {
+        return int_to_num(obj.value);
+    }
+    if (type(pyinterp.class_float)) {
+        return obj.value;
+    }
+    if (type(pyinterp.class_str)) {
+        return obj.value;
+    }
+    if (type(pyinterp.class_tuple)) {
+        return obj.seq.map(py2host);
+    }
+    if (type(pyinterp.class_list)) {
+        return obj.seq.map(py2host);
+    }
+    // TODO: dicts
+    if (type(pyinterp.class_bool)) {
+        return obj.value && true;
+    }
+    if (type(pyinterp.class_NoneType)) {
+        return null
+    }
+}
```

Comment:
should have a "catch all" here... like raising an exception

Reply:
See `throw`.

Reply:
where?